client/pluginmanager/drivermanager/manager_test.go

```diff
@@ -0,0 +1,300 @@
+package drivermanager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/nomad/client/pluginmanager"
+	"github.com/hashicorp/nomad/client/state"
+	"github.com/hashicorp/nomad/helper/testlog"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/hashicorp/nomad/plugins/base"
+	"github.com/hashicorp/nomad/plugins/drivers"
+	"github.com/hashicorp/nomad/plugins/shared/loader"
+	"github.com/hashicorp/nomad/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+var _ Manager = (*manager)(nil)
+var _ pluginmanager.PluginManager = (*manager)(nil)
+
+func testSetup(t *testing.T) (chan *drivers.Fingerprint, chan *drivers.TaskEvent, *manager) {
+	fpChan := make(chan *drivers.Fingerprint)
+	evChan := make(chan *drivers.TaskEvent)
+	drv := mockDriver(fpChan, evChan)
+	cat := mockCatalog(map[string]drivers.DriverPlugin{"mock": drv})
+	cfg := &Config{
+		Logger:       testlog.HCLogger(t),
+		Loader:       cat,
+		PluginConfig: nil,
+		Updater:      noopUpdater,
+		State:        state.NoopDB{},
+	}
```

Comment:
Hm, this doesn't fill all of the Config fields. Perhaps we should start a pattern of _always_ populating all Config fields in shared testSetup funcs and commenting why nil/zero values are ok?

I'm just concerned that our tests might vary from actual runtime conditions. It might also make future test failures due to missing Config fields a little easier to spot?

I'm not sure. Feel free to ignore.